### DISCUSSION TITLE
Why dataSchema in IcebergScanTransformer is empty?

### DISCUSSION THREAD

**Discussion by j7nhai** [apache/gluten#6409] [Q&A]
<img width="873" alt="image" src="https://github.com/apache/incubator-gluten/assets/146867566/86b9a205-c01a-4b5c-97a5-4c137a4e4c87">

@liujiayi771 Need some help : )

**Comment by liujiayi771** [comment]
@j7nhai You can check the invocation of this method, which is only used during validation. When this function was originally developed, it probably wasn't being used in Velox. Its usage in Velox should have been added later. When Iceberg support was initially developed, there should have been no need to construct a data schema.
